From e03a96dbb2d548890328e5be5170768e9747b428 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Mon, 3 Feb 2025 16:09:25 +0100
Subject: [PATCH] arm64: dts: st: change power domain for i2s on
 stm32mp251.dtsi

The SPI/I2S peripheral supports LP-Stop1/2 low power mode.
(See "Functionalities depending on system operating mode" table in ref
manual). However, the is2 driver does not support the autonomous mode.
Change power domain to "CLUSTER_PD" for i2s nodes, on stm32mp251.dtsi.

Change-Id: Ia5145598104da243090e613460c693b124fc27ef
Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/434539
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 3140d5436ebf..9efe77aa55d8 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -814,7 +814,7 @@ i2s2: audio-controller@400b0000 {
 					<&hpdma 52 0x43 0x21>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 23>;
-				power-domains = <&RET_PD>;
+				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 			};
 
@@ -846,7 +846,7 @@ i2s3: audio-controller@400c0000 {
 					<&hpdma 54 0x43 0x21>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 24>;
-				power-domains = <&RET_PD>;
+				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 			};
 
@@ -1250,7 +1250,7 @@ i2s1: audio-controller@40230000 {
 					<&hpdma 50 0x43 0x21>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 22>;
-				power-domains = <&RET_PD>;
+				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
 			};
 
-- 
2.50.1

